src/app/(main)/my-voyage/[teamId]/directory/components/DirectoryComponentWrapper.tsx

```diff
@@ -0,0 +1,135 @@
+import { redirect } from "next/navigation";
+import DirectoryProvider from "./DirectoryProvider";
+import TeamMember from "./TeamMember";
+import { TeamDirectory } from "@/store/features/directory/directorySlice";
+import Banner from "@/components/banner/Banner";
+import { getAccessToken } from "@/utils/getCookie";
+import { AsyncActionResponse, handleAsync } from "@/utils/handleAsync";
+import { GET } from "@/utils/requests";
+import { CacheTag } from "@/utils/cacheTag";
+import { User, VoyageTeamMember } from "@/store/features/user/userSlice";
+import { getUser } from "@/utils/getUser";
+import { getTimezone } from "@/utils/getTimezone";
+import VoyagePageBannerContainer from "@/components/banner/VoyagePageBannerContainer";
+
+interface FetchTeamDirectoryProps {
+  teamId: number;
+  user: User | null;
+}
+
+export async function fetchTeamDirectory({
+  teamId,
+  user,
+}: FetchTeamDirectoryProps): Promise<AsyncActionResponse<TeamDirectory>> {
+  const token = getAccessToken();
+
+  const fetchTeamDirectoryAsync = () =>
+    GET<TeamDirectory>(
+      `api/v1/teams/${teamId}`,
+      token,
+      "force-cache",
+      CacheTag.directory,
+    );
+
+  const [res, error] = await handleAsync(fetchTeamDirectoryAsync);
+
+  if (res) {
+    updateDirectoryWithCurrentTime(res);
+    const teamMember = res.voyageTeamMembers;
+    const elementToSort = teamMember.find(
+      (element) => element.member.discordId === user?.discordId,
+    );
+    moveElementToFirst(teamMember, elementToSort);
+  }
+
+  return [res, error];
+}
+
+function updateDirectoryWithCurrentTime(data: TeamDirectory) {
+  return data.voyageTeamMembers.forEach((teamMember) => {
+    const { timezone } = teamMember.member;
+    const currentTime = getTimezone(timezone);
+    teamMember.member.currentTime = currentTime;
+  });
+}
+
+function moveElementToFirst<T>(arr: T[], element: T): T[] {
+  const index = arr.indexOf(element);
+  if (index === -1) {
+    return arr;
+  }
+  [arr[index], arr[0]] = [arr[0], arr[index]];
+  return arr;
+}
+
+interface TeamDirectoryProps {
+  params: {
+    teamId: string;
+  };
+}
+
+export default async function DirectoryComponentWrapper({
+  params,
+}: TeamDirectoryProps) {
+  let teamDirectory: TeamDirectory;
+  let currentVoyageTeam: VoyageTeamMember | undefined;
+
+  const [user, error] = await getUser();
```

Comment:
@Dan-Y-Ko Can we avoid calling `getUser` again? I mean we've already made this call in `layout.tsx` and then set links in `Sidebar.tsx`. I don't understand what are you trying to check here:
`const teamId = currentVoyageTeam?.voyageTeamId && +params.teamId;`

Reply:
Oh, I made the changes in ideationcomponentwrapper...oops

Reply:
@Dan-Y-Ko Yeah, makes sense now. So users can't access their previous voyages (at least not by using `my-voyage` routes), got it 👍

Reply:
@JaneMoroz I pushed up a slightly refactored version of the getCurrentVoyageData function. Let me know what you think, and if it's worth keeping this. I personally don't think it provides much value and doesn't really save any code.